### PR TITLE
Migrate to RFG, fix memory leaks in models

### PR DESCRIPTION
**Pull request by jchung01**
I was having trouble setting up the dev environment with the old buildscript because of the deps, so I figured it would be better to just migrate it to a more modern version. Buildscript uses RetroFuturaGradle, based off of [TemplateDevEnv](https://github.com/CleanroomMC/TemplateDevEnv). This should make development more convenient.
**To setup the dev env**, you need to:
- Set your Gradle JVM to Java 17 or higher in IntelliJ -> File -> Settings -> Build, Execution, Deployment -> Build Tools -> Gradle  
- Run `setupDecompWorkspace` under the `modded minecraft` task category and refresh Gradle.

(See `dependencies.gradle` and `gradle.properties` for most things to change with the buildscript).

Main thing I wanted to fix here was memory leaks from ModelAnimator. There are quite a few static (or otherwise persistent) references in random places ([example](https://github.com/Rebirth-of-the-Night/Ice_and_Fire/blob/1.8.4-1.12.2/src/main/java/com/github/alexthe666/iceandfire/ClientProxy.java#L102-L105)) that end up retaining ModelAnimator, and because ModelAnimator has an `Entity` field, we need to make sure the field is nulled anytime after we update it and are done with that specific entity. 
**This is a cherry-picked commit from RLCraft's fork** (see original commit [here](https://github.com/kotlin-programmer/Ice_and_Fire_RLCRaft/commit/d725fe0ed61fb55dd8a2ade7f69cf7218880b31b)). I can provide proof of the memory leaks if needed, they were retaining `WorldClient`s which are usually quite large (several hundred MBs).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,122 +1,346 @@
...
+    if (propertyBool('generate_javadocs_jar')) {
+        withJavadocJar()
     }
 }
-apply plugin: 'net.minecraftforge.gradle.forge'
-//Only edit below this line, the above code adds and enables the nessasary things for Forge to be setup.
 
-version = "1.9.1-1.3.3"
-group= "iceandfire"
-archivesBaseName = "IceAndFireRotN"
```
Comment (democat3457) [inline build.gradle:26]: Is this property non-existent in RFG? (if not, how does RFG decide the archive base name?)

Comment (jchung01) [reply build.gradle:26]: I _believe_ it's set to the mod_id (see lines 47-49). The other two properties are determined by `mod_version` and `root_package` in `gradle.properties`. What does `archivesBaseName` usually control the name of?

Comment (democat3457) [reply build.gradle:26]: It's just convenience, changes the name of the produced jar - so building gives a jar with name `IceAndFireRotN-${version}.jar`

Comment (jchung01) [reply build.gradle:26]: Oops yeah, now I see it determines the build jar's name. Added the property to `gradle.properties`.

**Comment by jchung01** [comment]
Also probably worth mentioning any future entity models that get added should also make sure to call `animator.update(null)` at the end of the `animate()` method. (ModelGhost in @KELETU66666's PR?)

**Review by democat3457 — APPROVED** [review]

**Pull request merged**